### PR TITLE
Consistently throw MissingRequiredConfigurationException from MeterRegistry implementations

### PR DESCRIPTION
**Pull request by izeye**
This PR changes to fail fast when project ID which is mandatory is missing in Stackdriver meter registry.

### CODE REVIEW

File: implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverMeterRegistry.java [hunk 1]
```diff
@@ -87,6 +88,8 @@ public StackdriverMeterRegistry(StackdriverConfig config, Clock clock) {
     private StackdriverMeterRegistry(StackdriverConfig config, Clock clock, ThreadFactory threadFactory,
                                      Callable<MetricServiceSettings> metricServiceSettings) {
         super(config, clock);
+        requireNonNull(config.projectId());
```
Comment (shakuzen) [inline implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverMeterRegistry.java]: We should probably use the `MissingRequiredConfigurationException` class we have like at least `WavefrontMeterRegistry` does for #417.

Comment (izeye) [reply implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverMeterRegistry.java]: @shakuzen Thanks for the feedback! I created https://github.com/micrometer-metrics/micrometer/pull/1277 to resolve it against the 1.0.x branch. I'll push another commit on this PR for the 1.1.x branch.

Comment (izeye) [reply implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverMeterRegistry.java]: @shakuzen I pushed e59137b.

**Comment by shakuzen** [comment]
I think we'll need some changes (at least in tests) in Spring Boot now that some repositories fail fast on required properties that were not before. I have applied similar changes to the remaining repositories in 1.1.x that have required properties.

**Pull request merged**